app/models/katello/root_repository.rb

```diff
@@ -85,6 +85,10 @@ class RootRepository < Katello::Model
       :in => HTTP_PROXY_POLICIES,
       :message => _("must be one of the following: %s") % HTTP_PROXY_POLICIES.join(', ')
     }
+    validates :required_tags, allow_nil: true, format: {
```

Comment:
hmm I i'd think required_tags will be an array (like `:docker_tags_whitelist`). Any particular reason why you want to store it as string?

Reply:
That's the way Candlepin takes it in - a comma-separated string with no spaces.  It's stored in that format both in the product certificate and in the `requiredTags` value in the object Candlepin returns.

Reply:
use serializable like -> https://github.com/Katello/katello/blob/master/app/models/katello/root_repository.rb#L5

Reply:
```
=> {"content"=>
  {"created"=>"2020-10-12T19:41:32+0000",
   "updated"=>"2020-10-12T19:41:32+0000",
   "uuid"=>"4028face7504ef7e01751e5510ca01b0",
   "id"=>"1602531692751",
   "type"=>"yum",
   "label"=>"Default_Organization_SomeProduct_qwer",
   "name"=>"qwer",
   "vendor"=>"Custom",
   "contentUrl"=>"/custom/SomeProduct/qwer",
   "requiredTags"=>"rhel-7-workstation,rhel-7-server",
   "gpgUrl"=>nil,
   "modifiedProductIds"=>[],
   "arches"=>nil,
   "requiredProductIds"=>[],
   "metadataExpire"=>1,
   "releaseVer"=>nil},
 "enabled"=>true}

```

Reply:
But that is an idiosyncrasy related from Katello to Candlepin  right and not hammer/ui -> katello ? Are you saying candlepin requiring comma separated in requiredtags forces us in someway ?  Couldnt we just `requiredTags.join(',')` ?

Reply:
No it's not forced, I just thought it made sense to store it in the same format as cp.  I don't really feel strongly so I'll go ahead and change it